src/prefect/settings/models/server/database.py

```diff
@@ -146,30 +194,54 @@ class ServerDatabaseSettings(PrefectBaseSettings):
-        validation_alias=AliasChoices(
-            AliasPath("sqlalchemy_pool_size"),
-            "prefect_server_database_sqlalchemy_pool_size",
-            "prefect_sqlalchemy_pool_size",
-        ),
-    )
-
     connection_app_name: Optional[str] = Field(
         default=None,
-        description="Controls the application_name field for connections opened from the connection pool when using a PostgreSQL database with the Prefect API.",
+        description="Controls the application_name field for connections opened from the connection pool when using a PostgreSQL database with the Prefect backend.",
     )
 
-    sqlalchemy_max_overflow: Optional[int] = Field(
-        default=None,
-        description="Controls maximum overflow of the connection pool when using a PostgreSQL database with the Prefect API. If not set, the default SQLAlchemy maximum overflow value will be used.",
-        validation_alias=AliasChoices(
-            AliasPath("sqlalchemy_max_overflow"),
-            "prefect_server_database_sqlalchemy_max_overflow",
-            "prefect_sqlalchemy_max_overflow",
-        ),
-    )
+    # handle deprecated fields
+
+    def __getattribute__(self, name: str) -> Any:
+        if name in ["sqlalchemy_pool_size", "sqlalchemy_max_overflow"]:
+            warnings.warn(
+                f"Setting {name} has been moved to the `sqlalchemy` settings group.",
+                DeprecationWarning,
+            )
+            field_name = name.replace("sqlalchemy_", "")
+            return getattr(super().__getattribute__("sqlalchemy"), field_name)
+        return super().__getattribute__(name)
+
+    # validators
+
+    @model_validator(mode="before")
+    @classmethod
+    def set_deprecated_sqlalchemy_settings_on_child_model_and_warn(
+        cls, values: dict[str, Any]
+    ) -> dict[str, Any]:
+        """
+        Set deprecated settings on the child model.
+        """
+        # Initialize sqlalchemy settings if not present
+        if "sqlalchemy" not in values:
+            values["sqlalchemy"] = SQLAlchemySettings()
+
+        if "sqlalchemy_pool_size" in values:
+            warnings.warn(
+                "`sqlalchemy_pool_size` has been moved to the `sqlalchemy` settings group as `pool_size`.",
+                DeprecationWarning,
+            )
+            if "pool_size" not in values["sqlalchemy"].model_fields_set:
+                values["sqlalchemy"].pool_size = values["sqlalchemy_pool_size"]
+
+        if "sqlalchemy_max_overflow" in values:
+            warnings.warn(
+                "`sqlalchemy_max_overflow` has been moved to the `sqlalchemy` settings group as `max_overflow`.",
+                DeprecationWarning,
+            )
+            if "max_overflow" not in values["sqlalchemy"].model_fields_set:
+                values["sqlalchemy"].max_overflow = values["sqlalchemy_max_overflow"]
+
+        return values
```

Comment:
first part handles access to the values via the parent, second handles setting the values if passed to the parent. env vars should go straight to the child via `validation_alias` on the child model's fields